pages/base.py

```diff
@@ -52,40 +57,95 @@ def __getattr__(self, element):
+        'registries_link': (By.XPATH, '//nav[@id="navbarScope"]/div/div[@class="navbar-header"]/div[@class="dropdown"]/ul[@role="menu"]/li/a[@href="' + settings.OSF_HOME + '/registries/"]'),
+        'meetings_link': (By.XPATH, '//nav[@id="navbarScope"]/div/div[@class="navbar-header"]/div[@class="dropdown"]/ul[@role="menu"]/li/a[@href="' + settings.OSF_HOME + '/meetings/"]'),
+        'search_link': (By.XPATH, '//div[@id="secondary-navigation"]/ul/li/a[@href="' + settings.OSF_HOME + '/search/"]'),
+        'support_link': (By.XPATH, '//div[@id="secondary-navigation"]/ul/li/a[@href="/support/"]'),
+        'donate_link': (By.XPATH, '//div[@id="secondary-navigation"]/ul/li/a[@href="https://cos.io/donate"]'),
+        'user_dropdown': (By.CSS_SELECTOR, '#secondary-navigation > ul > li:nth-child(5) > button'),
+        'user_dropdown_profile': (By.XPATH, '//div[@id="secondary-navigation"]/ul/li[@class="dropdown"]/ul/li/a[@href="/logout/"]'),
+        'user_dropdown_support': (By.XPATH, '//div[@id="secondary-navigation"]/ul/li[@class="dropdown"]/ul/li/a[@href="' + settings.OSF_HOME + '/support/"]'),
+        'user_dropdown_settings': (By.XPATH, '//div[@id="secondary-navigation"]/ul/li[@class="dropdown"]/ul/li/a[@href="/settings/"]'),
+        'sign_up_button': (By.LINK_TEXT, 'Sign Up'),
+        'sign_in_button': (By.LINK_TEXT, 'Sign In'),
+        'logout_link': (By.CSS_SELECTOR, '#secondary-navigation > ul > li.dropdown.open > ul > li:nth-child(4) > a'),
+        'current_service': (By.CSS_SELECTOR, '#navbarScope > div > div > div.service-home > a > span.current-service > strong')
+    }
 
     def __init__(self, driver):
-        super(OSFBasePage, self).__init__(driver)
-        self.navbar = self.Navbar(driver)
+        super(Navbar, self).__init__(driver)
+
+    def verify(self):
+        return len(self.find_elements(By.XPATH, '//nav[@id="navbarScope"]')) == 1
 
     def is_logged_in(self):
-        return self.navbar.is_logged_in()
+        try:
+            if self.sign_in_button:
+                return False
+        except ValueError:
+            return True
 
 
-    class Navbar(BaseElement):
+class OSFBasePage(BasePage):
+    url = settings.OSF_HOME
 
-        locators = {
-            'sign_in_button': (By.LINK_TEXT, 'Sign In'),
-            'user_dropdown': (By.CSS_SELECTOR, '#secondary-navigation > ul > li:nth-child(5) > button'),
-            'logout_link': (By.CSS_SELECTOR, '#secondary-navigation > ul > li.dropdown.open > ul > li:nth-child(4) > a')
-        }
+    locators = {
+        'error_heading': (By.CSS_SELECTOR, 'h2#error')
+    }
 
-        def is_logged_in(self):
+    def __init__(self, driver, goto=True):
+        super(OSFBasePage, self).__init__(driver)
+        if goto:
+            # Verify the page is what you expect it to be.
+            driver.get(self.url)
+
+        if not self.verify():
```

Comment:
I like the idea of verifying the page when you instantiate the page object, but unfortunately I don't think it works in practice. The way it is written now, if any page object doesn't have a verify method it will return true, meaning that people will have to go and double check if people want to call the verify to see if it's actually implemented before using it.